samples/Common/cs/FrameReaderFrameSource.cs

```diff
@@ -177,6 +191,33 @@ private async Task InitializeMediaFrameSourceAsync()
             FrameHeight = m_frameSource.CurrentFormat.VideoFormat.Height;
         }
 
+        /// <summary>
+        /// Return a MediaEncodingSubtype string corrolary to a BitmapPixelFormat
+        /// </summary>
+        /// <param name="supportedBitmapPixelFormat"></param>
+        /// <returns></returns>
+        private string BitmapPixelFormatToMediaEncodingSubtype(BitmapPixelFormat supportedBitmapPixelFormat)
+        {
+            switch(supportedBitmapPixelFormat)
+            {
+                case BitmapPixelFormat.Nv12:
+                    return MediaEncodingSubtypes.Nv12;
+                case BitmapPixelFormat.Yuy2:
+                    return MediaEncodingSubtypes.Yuy2;
+                case BitmapPixelFormat.Gray8:
+                    return MediaEncodingSubtypes.L8;
+                case BitmapPixelFormat.Gray16:
+                    return MediaEncodingSubtypes.L16;
+                case BitmapPixelFormat.P010:
+                    return MediaEncodingSubtypes.P010;
+                case BitmapPixelFormat.Rgba8:
+                case BitmapPixelFormat.Bgra8:
+                    return MediaEncodingSubtypes.Bgra8;
+            }
+            // By default return BGRA8
```

Comment:
nit: why not just use default case in the switch statement?

Reply:
I have to return a string in this method. If I use default, I make this final result useless. I save a line of code by returning the default value at all time if not caught in a switch case.

---
In reply to: [301348544](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/46#discussion_r301348544) [](ancestors = 301348544)